include/seqan3/io/sam_file/format_bam.hpp

```diff
@@ -961,6 +945,37 @@ inline void format_bam::write_alignment_record([[maybe_unused]] stream_type & st
     } // if constexpr (!detail::decays_to_ignore_v<header_type>)
 }
 
+template <typename stream_t, typename ref_ids_type>
+inline void format_bam::write_header(stream_t & stream,
+                                     sam_file_output_options const & options,
+                                     sam_file_header<ref_ids_type> & header)
+{
+    detail::fast_ostreambuf_iterator stream_it{*stream.rdbuf()};
+
+    std::ranges::copy_n("BAM\1", 4, stream_it);
+    std::ostringstream os;
+    detail::format_sam_base::write_header(os, options, header); // write SAM header to temporary stream to query the size.
+    int32_t l_text{static_cast<int32_t>(os.str().size())};
```

Comment:
```suggestion
    int32_t const l_text{static_cast<int32_t>(os.view().size())};
```

`std::ostringstream::view()` is CPP20, let's hope, GCC 10 implements it